zebra-chain/src/sapling/tree.rs

```diff
@@ -72,37 +82,181 @@ impl fmt::Debug for Root {
     }
 }
 
-impl From<SaplingNoteCommitmentTree> for Root {
-    fn from(_tree: SaplingNoteCommitmentTree) -> Self {
-        // TODO: The Sapling note commitment tree requires a Pedersen
-        // hash function, not SHA256.
-
-        // let mut hash_writer = Sha256dWriter::default();
-        // sapling_note_commitment_tree
-        //     .zcash_serialize(&mut hash_writer)
-        //     .expect("A Sapling note commitment tree must serialize.");
-        // Self(hash_writer.finish())
+/// Sapling Note Commitment Tree
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
+struct NoteCommitmentTree {
+    /// The root node of the tree (often used as an anchor).
+    root: Root,
+    /// The height of the tree (maximum height for Sapling is 32).
+    height: u8,
```

Comment:
We probably want to use a custom strategy here so we don't generate invalid heights

https://altsysrq.github.io/proptest-book/proptest-derive/modifiers.html#strategy

Reply:
Actually I might remove the `Arbitrary` for `NoteCommitmentTree` as  it's basically an intermediate computation structure, having it for `Root` etc is useful though.

Reply:
https://github.com/ZcashFoundation/zebra/pull/1045/commits/999d946ff4225798a6a17ef10624425814cfe430